### DISCUSSION TITLE
Any way to use macOS preview?

### DISCUSSION THREAD

**Discussion by kumamaki** [tauri-apps/tauri#9660] [Q&A]
Is there anyway to use MacOS preview functionality? I did some research and seems like there're no crates.  
The only option is `qlmanage` which is just a testing tool and doesn't satisfy my needs.

**Comment by FabianLars** [comment]
Did you try using the shell.open api already? Since Preview is the default viewer for images etc it may be a good starting point.
https://tauri.app/v1/api/js/shell/#open / https://crates.io/crates/open (the crate also allows you to basically do `open -a Preview filename.ext` using `open::with`).

**Comment by kumamaki** [comment]
Oh yeah, my bad! I'm talking about Quick Look, not the Preview app.

**Reply by FabianLars** [reply]
Hmm, there indeed doesn't seem to be an alternative to `qlmanage -p` :/

**Reply by kumamaki** [reply]
Yeah. But the fact that it's possible with Swift, means it's possible with a Rust binding?

**Reply by FabianLars** [reply]
I just noticed i never pressed Send back then 😭 

What did you mean with it's "possible with Swift"?
if you mean the qlmanage cli itself then you'd have to reverse engineer that somehow because it uses apis that seemingly aren't documented.
i know that i saw some swift apis that looked related but actually weren't - can't remember them 2 months later though :/

**Reply by FabianLars** [reply]
Ah no, i think i remember, it were the QLPreview apis, right? As far as i understood it, that makes it possible to preview something inside your own app window. If that's what you want, then i guess it'd be possible.

You can use swift via swift-rs or objective-c via objc2 - maybe this example will also be helpful, idk just saw it on SO somewhere https://developer.apple.com/library/archive/samplecode/QuickLookDownloader/Introduction/Intro.html#//apple_ref/doc/uid/DTS40009082-Intro-DontLinkElementID_2

**Reply by josmanvis** [reply]
2026 and still no crates :(
